### ISSUE TITLE
Failing with no violation found but templating errors

### ISSUE THREAD

**Issue by landier** [sqlfluff/sqlfluff#2202]
SQLFluff is failing without finding any violation nor fixing any file but finds `templating error`. What are those?

## Expected Behaviour
Since SQLFluff is not touching the file I'm running `pre-commit run --files` against, I would not expect it to fail.

## Observed Behaviour
```
trim trailing whitespace.................................................Passed
fix end of files.........................................................Passed
check yaml...........................................(no files to check)Skipped
check for added large files..............................................Passed
sqlfluff-fix.............................................................Failed
- hook id: sqlfluff-fix
- exit code: 1

==== finding fixable violations ====
==== no fixable linting violations found ====
All Finished 📜 🎉!
  [2 templating errors found]
```

## Steps to Reproduce
Given a file like this one:
```SQL
SELECT
    CURRENT_DATE() AS day,
    {{ dbt_utils.star(from=source('...', '....')) }}
FROM
    {{ source('...', '....') }}
```

## Dialect
Snowflake

## Version
Sqlfluff pre-commit hook: 0.9.0
DBT: 1.0.0

## Configuration
```
[sqlfluff]
; Disable some rules: https://docs.sqlfluff.com/en/stable/rules.html
; L016: Line is too long
; L031: Avoid table aliases in from clauses and join conditions.
; L034: Use wildcards then simple select targets before calculations and aggregates.
; L043: Unnecessary case when statement. Use the “when” condition itself.
exclude_rules = L016,L031,L034,L043
dialect = snowflake

[sqlfluff:rules:L010]
capitalisation_policy = upper

[sqlfluff:rules:L014]
extended_capitalisation_policy = lower
unquoted_identifiers_policy = all

```

**Comment by tunetheweb** [comment]
Run `sqfluff lint` on the file and it’ll show the templating lines it’s struggling with.

However I would guess the issue is that you seem to be using dbt, but have not set the templater from the default of `jinja` to `dbt` in your config.

**Comment by landier** [comment]
Thanks. Sorry for the dumb question, I do not use sqlfluff directly but only via the pre-commit hook.

Indeed the `sqlfluff lint` told me more and it's actually failing because of DBT stuff in the Jinja template:
```
L:   0 | P:   0 |  TMP | Unrecoverable failure in Jinja templating: 'dbt_utils'
                       | is undefined. Have you configured your variables?
                       | https://docs.sqlfluff.com/en/latest/configuration.html
L:  13 | P:  16 |  TMP | Undefined jinja template variable: 'dbt_utils'
```

Unfortunately, I can't switch to the DBT templater, it's too tigthly coupled with a local DBT. That breaks our setup.

Thanks again!
